src/pages/RecipeFinder.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+
+import generateRecipe from '../api/OpenAIAPI.js';
+
+export default function RecipeFinder() {
+  const [ingredients, setIngredients] = useState('');
+  const [recipe, setRecipe] = useState(null);
+
+  function handleInputChange(event) {
+    setIngredients(event.target.value);
+  };
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    try {
+      const response = await generateRecipe(ingredients);
+      setRecipe(response);
+    } catch (error) {
+      console.error('Error:', error);
+    }
+  };
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          placeholder="Enter ingredients"
+          value={ingredients}
+          onChange={handleInputChange}
+          required
+        />
+        <button type="submit">Find Recipe</button>
+      </form>
+      {recipe && (
+        <div>
+          <h2>Recipe</h2>
```

Comment:
**suggestion:** At some point you should consider the structure of the site. Where h1 is the main header and h2's would be meaningful sub headers. Try to shy away from using headlines just for their style. The main purpose of them is for accessibility and web crawlers.